packages/sync/src/modules/class-full-sync-immediately.php

```diff
@@ -0,0 +1,395 @@
+<?php
+/**
+ * Full sync module.
+ *
+ * @package automattic/jetpack-sync
+ */
+
+namespace Automattic\Jetpack\Sync\Modules;
+
+use Automattic\Jetpack\Sync\Defaults;
+use Automattic\Jetpack\Sync\Lock;
+use Automattic\Jetpack\Sync\Modules;
+use Automattic\Jetpack\Sync\Settings;
+
+/**
+ * This class does a full resync of the database by
+ * sending an outbound action for every single object
+ * that we care about.
+ */
+class Full_Sync_Immediately extends Module {
+	/**
+	 * Prefix of the full sync status option name.
+	 *
+	 * @var string
+	 */
+	const STATUS_OPTION = 'jetpack_sync_full_status';
+
+	/**
+	 * Sync Lock name.
+	 *
+	 * @var string
+	 */
+	const LOCK_NAME = 'full_sync';
+
+	/**
+	 * Sync module name.
+	 *
+	 * @access public
+	 *
+	 * @return string
+	 */
+	public function name() {
+		return 'full-sync';
+	}
+
+	/**
+	 * Initialize action listeners for full sync.
+	 *
+	 * @access public
+	 *
+	 * @param callable $callable Action handler callable.
+	 */
+	public function init_full_sync_listeners( $callable ) {
+	}
+
+	/**
+	 * Start a full sync.
+	 *
+	 * @access public
+	 *
+	 * @param array $full_sync_config Full sync configuration.
+	 *
+	 * @return bool Always returns true at success.
+	 */
+	public function start( $full_sync_config = null ) {
+		// There was a full sync in progress.
+		if ( $this->is_started() && ! $this->is_finished() ) {
+			/**
+			 * Fires when a full sync is cancelled.
+			 *
+			 * @since 4.2.0
+			 */
+			do_action( 'jetpack_full_sync_cancelled' );
+			$this->send_action( 'jetpack_full_sync_cancelled' );
+		}
+
+		// Remove all evidence of previous full sync items and status.
+		$this->reset_data();
+
+		if ( ! is_array( $full_sync_config ) ) {
+			$full_sync_config = Defaults::$default_full_sync_config;
+			if ( is_multisite() ) {
+				$full_sync_config['network_options'] = 1;
+			}
+		}
+
+		if ( isset( $full_sync_config['users'] ) && 'initial' === $full_sync_config['users'] ) {
+			$full_sync_config['users'] = Modules::get_module( 'users' )->get_initial_sync_user_config();
+		}
+
+		$this->update_status(
+			array(
+				'started'  => time(),
+				'config'   => $full_sync_config,
+				'progress' => $this->get_initial_progress( $full_sync_config ),
+			)
+		);
+
+		$range = $this->get_content_range( $full_sync_config );
+		/**
+		 * Fires when a full sync begins. This action is serialized
+		 * and sent to the server so that it knows a full sync is coming.
+		 *
+		 * @param array $full_sync_config Sync configuration for all sync modules.
+		 * @param array $range Range of the sync items, containing min and max IDs for some item types.
+		 * @param array $empty The modules with no items to sync during a full sync.
+		 *
+		 * @since 4.2.0
+		 * @since 7.3.0 Added $range arg.
+		 * @since 7.4.0 Added $empty arg.
+		 */
+		do_action( 'jetpack_full_sync_start', $full_sync_config, $range );
```

Comment:
I gather we're using the same actions because this one is aiming to replace the other full sync, right?

Reply:
right

Reply:
I leave it here just in case someone is listening to it, but it is not needed anymore for Full Sync purposes.